File: examples/android/MSDKSampleAppJava/lib/src/main/java/msdk/java/types/ProvisioningConfig.java
package msdk.java.types;

import androidx.annotation.NonNull;
import androidx.annotation.RawRes;

import org.json.JSONException;
import org.json.JSONObject;

import msdk.java.handlers.Initialization;

public class ProvisioningConfig {
    public static ProvisioningConfigBuilder builder() {
        return new ProvisioningConfigBuilder();
    }

    public static final class ProvisioningConfigBuilder {
        private String agencyEndpoint;
        private String agencyDid;
        private String agencyVerkey;
        private String walletName;
        private String walletKey;
        private String logo;
        private String name;

        private ProvisioningConfigBuilder() {
        }

        /**
         * Set agency endpoint.
         *
         * @param agencyEndpoint agency endpoint
         * @return {@link ProvisioningConfigBuilder} instance
         */
        public @NonNull
        ProvisioningConfigBuilder withAgencyEndpoint(@NonNull String agencyEndpoint) {
            this.agencyEndpoint = agencyEndpoint;
            return this;
        }

        /**
         * Set agency did.
         *
         * @param agencyDid agency did
         * @return {@link ProvisioningConfigBuilder} instance
         */
        public @NonNull
        ProvisioningConfigBuilder withAgencyDid(@NonNull String agencyDid) {
            this.agencyDid = agencyDid;
            return this;
        }

        /**
         * Set agency verkey.
         *
         * @param agencyVerkey agency verkey
         * @return {@link ProvisioningConfigBuilder} instance
         */
        public @NonNull
        ProvisioningConfigBuilder withAgencyVerkey(@NonNull String agencyVerkey) {
            this.agencyVerkey = agencyVerkey;
            return this;
        }

        /**
         * Set wallet name.
         *
         * @param walletName wallet name
         * @return {@link ProvisioningConfigBuilder} instance
         */
        public @NonNull
        ProvisioningConfigBuilder withWalletName(@NonNull String walletName) {
            this.walletName = walletName;
            return this;
        }

        /**
         * Set logo
         *
         * @param logo logo
         * @return this
         */
        public @NonNull
        ProvisioningConfigBuilder withLogo(String logo) {
            this.logo = logo;
            return this;
        }

        /**
         * Set app name
         *
         * @param name name
         * @return this
         */
        public @NonNull
        ProvisioningConfigBuilder withName(String name) {
            this.name = name;
            return this;
        }

        /**
         * Set wallet key
         *
         * @param walletKey walletKey
         * @return this
         */
        public @NonNull
        ProvisioningConfigBuilder withWalletKey(String walletKey) {
            this.walletKey = walletKey;
            return this;
        }

        /**
         * Creates provisioning config from settings.
         *
         * @return provisioning config
         */
        public @NonNull
        String build() throws JSONException {
            JSONObject agencyConfig = new JSONObject();
            agencyConfig.put("agency_endpoint", this.agencyEndpoint);
            agencyConfig.put("agency_did", this.agencyDid);
            agencyConfig.put("agency_verkey", this.agencyVerkey);
            agencyConfig.put("wallet_name", this.walletName);
            agencyConfig.put("wallet_key", this.walletKey);
            agencyConfig.put("protocol_type", "3.0");
            agencyConfig.put("logo", this.logo);
            agencyConfig.put("name", this.name);
            return agencyConfig.toString();
        }
    }
}

File: examples/android/MSDKSampleAppKotlin/app/src/main/java/msdk/kotlin/sample/types/ProvisioningConfig.kt
package msdk.kotlin.sample.types

import org.json.JSONException
import org.json.JSONObject


object ProvisioningConfig {
    fun builder(): ProvisioningConfigBuilder {
        return ProvisioningConfigBuilder()
    }

    class ProvisioningConfigBuilder {
        private var agencyEndpoint: String? = null
        private var agencyDid: String? = null
        private var agencyVerkey: String? = null
        private var walletName: String? = null
        private var walletKey: String? = null
        private var logo: String? = null
        private var name: String? = null
        private var protocolType: String? = null

        /**
         * Set agency endpoint.
         *
         * @param agencyEndpoint agency endpoint
         * @return [ProvisioningConfigBuilder] instance
         */
        fun withAgencyEndpoint(agencyEndpoint: String): ProvisioningConfigBuilder {
            this.agencyEndpoint = agencyEndpoint
            return this
        }

        /**
         * Set agency did.
         *
         * @param agencyDid agency did
         * @return [ProvisioningConfigBuilder] instance
         */
        fun withAgencyDid(agencyDid: String): ProvisioningConfigBuilder {
            this.agencyDid = agencyDid
            return this
        }

        /**
         * Set agency verkey.
         *
         * @param agencyVerkey agency verkey
         * @return [ProvisioningConfigBuilder] instance
         */
        fun withAgencyVerkey(agencyVerkey: String): ProvisioningConfigBuilder {
            this.agencyVerkey = agencyVerkey
            return this
        }

        /**
         * Set wallet name.
         *
         * @param walletName wallet name
         * @return [ProvisioningConfigBuilder] instance
         */
        fun withWalletName(walletName: String): ProvisioningConfigBuilder {
            this.walletName = walletName
            return this
        }

        /**
         * Set logo
         *
         * @param logo logo
         * @return this
         */
        fun withLogo(logo: String?): ProvisioningConfigBuilder {
            this.logo = logo
            return this
        }

        /**
         * Set app name
         *
         * @param name name
         * @return this
         */
        fun withName(name: String?): ProvisioningConfigBuilder {
            this.name = name
            return this
        }

        /**
         * Set wallet key
         *
         * @param walletKey walletKey
         * @return this
         */
        fun withWalletKey(walletKey: String?): ProvisioningConfigBuilder {
            this.walletKey = walletKey
            return this
        }

        /**
         * Set protocol type
         *
         * @param protocolType protocol type
         * @return this
         */
        fun withProtocolType(protocolType: String?): ProvisioningConfigBuilder {
            this.protocolType = protocolType
            return this
        }

        /**
         * Creates provisioning config from settings.
         *
         * @return provisioning config
         */
        @Throws(JSONException::class)
        fun build(): String {
            val agencyConfig = JSONObject()
            agencyConfig.put("agency_endpoint", agencyEndpoint)
            agencyConfig.put("agency_did", agencyDid)
            agencyConfig.put("agency_verkey", agencyVerkey)
            agencyConfig.put("wallet_name", walletName)
            agencyConfig.put("wallet_key", walletKey)
            agencyConfig.put("protocol_type", "3.0")
            agencyConfig.put("logo", logo)
            agencyConfig.put("name", name)
            agencyConfig.put("protocol_type", this.protocolType);
            return agencyConfig.toString()
        }
    }
}